api/http/api_deployments.go

```diff
@@ -379,8 +378,16 @@ func (d *DeploymentsApiHandlers) DownloadConfiguration(w rest.ResponseWriter, r
 	hdr := rw.Header()
 	hdr.Set("Content-Disposition", `attachment; filename="artifact.mender"`)
 	hdr.Set("Content-Type", app.ArtifactContentType)
+	artifactPayload, err := ioutil.ReadAll(artifact)
+	if err != nil {
+		// There's not anything we can do here in terms of the response.
+		rw.WriteHeader(http.StatusOK)
+		l.Error(err.Error())
+		return
+	}
+	hdr.Set("Content-Length", strconv.Itoa(len(artifactPayload)))
```

Comment:
If you buffer the artifact in memory, it's not too late to report an internal error while reading it.
```suggestion

	artifactPayload, err := ioutil.ReadAll(artifact)
	if err != nil {
		d.view.RenderInternalError(e, r, err, l)
		return
	}
	hdr := rw.Header()
	hdr.Set("Content-Disposition", `attachment; filename="artifact.mender"`)
	hdr.Set("Content-Type", app.ArtifactContentType)
	hdr.Set("Content-Length", strconv.Itoa(len(artifactPayload)))
```

Reply:
On the other hand, I believe `net/http` adds the content length header if the response is written in a single write.

Reply:
Right, reworked. Regarding the content-length, it is not there if we don't set it explicitly...